### ISSUE TITLE
When the window is maximized or minimized, the bottom view will be covered

### ISSUE THREAD

**Issue by MaxBill** [r0x0r/pywebview#817]
### Specification
- pywebview version: 3.5
- operating system: windows 10 and windows 10
- web renderer:  cef and mshtml

### Description  

After the window instantiation is started, maximize or minimize the window or resize, and then restore the window. At this time, compared with the original, some views at the bottom are blocked. This is the case with CEF and mshtml. After browsing the documents and issues, this problem has not been solved. Please ask how to troubleshoot this problem. Thank you very much.

Here are two screenshots of the original and problem comparison:
![image](https://user-images.githubusercontent.com/16558525/142734387-a964ac61-5ef8-43b2-8f50-f6bf42073807.png)
![image](https://user-images.githubusercontent.com/16558525/142734411-3c2f9983-6fc7-472c-9fdd-80e8ea1ed660.png)

  
  
### Practicalities

- YES/NO I am willing to work on this issue myself.

- YES/NO I am prepared to support this issue financially.

**Comment by r0x0r** [comment]
Please share the code to reproduce this problem

**Comment by MaxBill** [comment]
I found a problem in the process of putting forward simple error examples from complex projects. This scenario occurs when the GUI is CEF, and mshtml has no problem. The following simple error examples I provide, thank you again for your help

```
import webview

html = '''
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="renderer" content="webkit">
    <title>hello pywebview</title>
    <style>
        * {
            margin: 0;
            padding: 0;
        }

        body {
            background-color: #3e8f3e;
        }

        iframe {
            border: 0;
        }

        #header {
            width: 100%;
            height: 90px;
            background-color: red;
        }

        #center {
            width: 100%;
            height: 100%;
            background-color: white;
        }

        #footer {
            bottom: 0;
            width: 100%;
            height: 30px;
            position: fixed;
            background-color: blue;
        }
    </style>
</head>
<body>
<div id="header">
   top
</div>
<div id="center">
    center
</div>
<div id="footer">
   bottom
</div>
</body>
</html>
'''


def run():
    window = webview.create_window(
        '', html=html,
        width=600, height=400
    )
    webview.start(gui='cef')


if __name__ == '__main__':
    run()

```

**Comment by MaxBill** [comment]
Can you repeat this question over there. Can the CEF version be upgraded? I wonder if this problem can be solved by upgrading CEF

**Comment by r0x0r** [comment]
Sorry haven't had a chance to investigate the problem yet. I think the problem is caused that CEF component fails to update its size on window restore event. CEF browser object provides NotifyMoveOrResizeStarted method. My guess is it should be invoked on maximize/restore events.

As for CEF upgrade. I haven't done it, but from my investigations back then I got an impression it was possible. CEF builds can be downloaded at https://cef-builds.spotifycdn.com/index.html and CEF Python documentation here https://github.com/cztomczak/cefpython

**Comment by MaxBill** [comment]
Thank you for paying attention to this problem. According to the solution ideas you provided, I will try to solve this problem. If there is a solution, I will share it here. I will close this problem first and thank you again for your help

**Comment by MaxBill** [comment]
Hello, through the debug code, I found that this problem is caused by a fixed value. When the window size changes, the incoming height parameter will be fixed minus 38. Please tell me what the value of 38 is. I try to change it to 44. On my computer, this problem is solved. I guess whether this 38 is the height of the taskbar, If so, there may be some compatibility problems with different versions of the system

![image](https://user-images.githubusercontent.com/16558525/144193365-61ff6813-b120-4199-89ed-e4dfef1abb5a.png)

**Comment by r0x0r** [comment]
It is the height of titlebar. It has been changed before to fix a similar problem, but apparently hard coding this value does not solve anything.

**Comment by r0x0r** [comment]
I pushed a potential fix to `cef-resize-fix` branch. It still has hard-coded values, but offset is calculated dynamically now. @MaxBill can you test if it solves the problem for you?

**Comment by r0x0r** [comment]
I merged the fix with 4.0 branch and will close this issue now.
